src/app/legacy/containers/Blocks/index.jsx

```diff
@@ -1,20 +1,34 @@
 import React, { Fragment } from 'react';
 import styled from '@emotion/styled';
 import path from 'ramda/src/path';
+import pathEq from 'ramda/src/pathEq';
 
 const Clearer = styled.div`
   clear: both;
 `;
 
-const Blocks = ({ blocks, componentsToRender }) =>
-  blocks.map((block, index) => {
+const Blocks = ({ blocks, componentsToRender }) => {
```

Comment:
I'd personally have the Related Content link display logic within the actual `JumpTo` component.

You could have an API with something like:
```
    jumpTo: (props: ComponentToRenderProps) => (
      <JumpTo {...props} showRelatedContentLink={showRelatedContent} />
    ),
```

`showRelatedContent` would be calculated within the ArticlePage component. It would mean you're not relying on `Blocks` to calculate and insert the related content block, leaving the data not mutated.

Reply:
I'll give this a go. I found the `jumpTo: JumpTo` along with the use of componentsToRender and <Blocks> component a bit confusing with how to pass anything into the jumpTo component! I am glad I can change that line 😁

Reply:
Also the linting is not liking me using useContext in Blocks as it is being used inside a callback

Reply:
It should make testing a lot simplier as well, as it will just be a case of `if showRelatedContentLink is true, should show link` etc, no need for finding it within the blocks or adding additional mock data.

Reply:
When working on this I am thinking, could I have all the logic actually in the ArticlePage? The reason I think this is because the thing that decides at the moment whether the related content link is shown or not is whether it is inserted into the jumpToHeadings array. When it is in there, it gets displayed in the same jumpToHeadings.map function that the other headings do (with only a small variance in the id used, where sending the boolean in can be useful to avoid needing to check translation in the jumpTo component) . When it isn't in there, it just doesn't display. Therefore the decision whether to include it happens at the point of insertion into the array. We can't do this inside the component because of the content shift on render. So I need to insert it into the array before the initial component render. I did this in the Blocks component before, but I can do it in the ArticlePage instead if I can access the array there.

As you suggest the showRelatedContent would be calculated in the ArticlePage, but we don't need to use that to decide whether to display it or not in the component, as it is decided when it is inserted into the array before the component renders.

Reply:
Its possible to do it in the ArticlePage, the problem is is that `blocks` will be mutated since you're using `.push()` to push in the related content into the `jumpToHeadings`. Ideally, we shouldn't mutate data directly. We can take a copy and then mutate, but that seems a bit expensive for this purpose.

Displaying the related content / most read based on a boolean flag detaches the logic from the blocks.

If you wanted to keep it in the `jumpToHeadings`, then I would do that logic in the BFF before the response gets to Simorgh.

Reply:
I have made this change thinking that there might be problem doing the translation inside the component with the text changing after render. But it is fine! I will push my changes soon.

Reply:
Cool, looks like its on the right track 👍